src/components/HyloTipTapEditor/HyloTipTapEditor.js

```diff
@@ -0,0 +1,213 @@
+  onAddTopic,
+  onBeforeCreate = () => {},
+  onChange,
+  onEnter,
+  onEscape,
+  placeholder,
+  readOnly
+}, ref) {
+  const dispatch = useDispatch()
+  const editorRef = useRef(null)
+  const [selectedLink, setSelectedLink] = useState()
+  const editor = useEditor({
+    content: contentHTML,
+
+    extensions: [
+      // Key events respond are last extension first, these will be last
+      Extension.create({
+        name: 'KeyboardShortcuts',
+        // Keep around for debugging for now:
+        // onTransaction: ({ editor, transaction }) => {
+        //   console.log('!!!!! looking how to get all link marks', transaction)
+        //   transactions.doc.node.forEach(child => {
+        //     const [fontSizeMark] = child.marks.filter((m: Mark) => m.type === markType)
+        //    })
+        // },
+        addKeyboardShortcuts () {
+          return {
+            Enter: ({ editor }) => {
+              if (!onEnter) return false
+              return onEnter(editor.getHTML())
+            },
+            Escape: () => {
+              if (!onEscape) return false
+              return onEscape()
+            }
+          }
+        }
+      }),
+
+      StarterKit.configure({
+        heading: {
+          levels: [1, 2, 3]
+        }
+      }),
+
+      Placeholder.configure({ placeholder }),
+
+      Link.extend({
+        // This expands concatenated links back to full href for editing
+        parseHTML () {
+          return [
+            {
+              tag: 'a',
+              // Special handling for links who's innerHTML has been concatenated by the backend
+              contentElement: element => {
+                if (element.innerHTML.match(/…$/)) {
+                  const href = element.getAttribute('href')
+
+                  try {
+                    const url = new URL(href)
+
+                    element.innerHTML = `${url.hostname}${url.pathname !== '/' ? url.pathname : ''}`
+                    return element
+                  } catch (e) {
+                    element.innerHTML = href
+                    return element
+                  }
+                }
+
+                return element
+              }
+            }
+          ]
+        },
+        addOptions () {
+          return {
+            ...this.parent?.(),
+            openOnClick: false,
+            autolink: true,
+            HTMLAttributes: {
+              target: null
+            },
+            validate: href => {
+              onAddLink && onAddLink(href)
+              return true
+            }
+          }
+        }
+      }),
+
+      PeopleMentions({ onSelection: onAddMention, maxSuggestions, groupIds, dispatch }),
+
+      TopicMentions({ onSelection: onAddTopic, maxSuggestions, groupIds, dispatch }),
+
+      Highlight
+    ],
+
+    onBeforeCreate,
+
+    onUpdate: ({ editor }) => {
+      if (
+        !onChange ||
+        // TODO: This condition won't run on last update in mobile. Test on Web with removal.
```

Comment:
anything TODO here now?

Reply:
Yes part of final bits I'm still on for mobile editor tonight 👍🏼